content/docs/reference-react-component.md

```diff
@@ -476,73 +476,73 @@ React doesn't call `UNSAFE_componentWillReceiveProps()` with initial props durin
 
-`setState()` will always lead to a re-render unless `shouldComponentUpdate()` returns `false`. If mutable objects are being used and conditional rendering logic cannot be implemented in `shouldComponentUpdate()`, calling `setState()` only when the new state differs from the previous state will avoid unnecessary re-renders.
+A `setState()` mindig újrarendereléshez vezet, kivéve, ha a `shouldComponentUpdate()` `false`-ot ad vissza. Ha változtatható objektumokat használsz és nem lehet implementálni feltételes renderelő logikát a `shouldComponentUpdate()`-ben, a felesleges renderelések elkerülése érdekében a `setState()`-et csak akkor hívd meg, ha az új állapot különbözik az előzőtől.
 
-The first argument is an `updater` function with the signature:
+Az első argumentum egy `updater` függvény az alábbi szignatúrával:
 
 ```javascript
 (state, props) => stateChange
 ```
 
-`state` is a reference to the component state at the time the change is being applied. It should not be directly mutated. Instead, changes should be represented by building a new object based on the input from `state` and `props`. For instance, suppose we wanted to increment a value in state by `props.step`:
+A `state` egy referencia a komponens állapotára a frissítés idejében. Ezt nem szabad közvetlenül módosítani. Ehelyett a változásokat reprezentáld egy új objektumban a `state` és `props` alapján. Például tegyük fel, hogy meg akarjuk növelni egy állapotváltozó értéket `props.step`-pel:
 
 ```javascript
 this.setState((state, props) => {
   return {counter: state.counter + props.step};
 });
 ```
 
-Both `state` and `props` received by the updater function are guaranteed to be up-to-date. The output of the updater is shallowly merged with `state`.
+Mind `state` és `props` értéke, amit a frissítőfüggvény (updater) által kaptunk, garantáltan friss lesz. A frissítőfüggvény kimenete a `state`-tel kerül sekélyen összefésülésre.
 
-The second parameter to `setState()` is an optional callback function that will be executed once `setState` is completed and the component is re-rendered. Generally we recommend using `componentDidUpdate()` for such logic instead.
+A `setState()` második paramétere egy opcionális callback függvény, ami akkor lesz meghívva, amikor a `setState` lefutott és a komponens újrarenderelt. Mi általában hasonló logikához a `componentDidUpdate()`-et ajánljuk ehelyett.
 
-You may optionally pass an object as the first argument to `setState()` instead of a function:
+A `setState()` első argumentuma függvény helyett lehet objektum is:
 
 ```javascript
 setState(stateChange[, callback])
 ```
 
-This performs a shallow merge of `stateChange` into the new state, e.g., to adjust a shopping cart item quantity:
+Ez a `stateChange` egy sekély összefésülését végzi el az új állapotba, pl., amikor egy bevásárlókocsi elemeinek számát változtatod meg:
```

Comment:
```suggestion
Ez a `stateChange` egy sekély összefésülését végzi el az új állapotba, pl., amikor egy kosár elemeinek számát változtatod meg:
```

On the web, I think we use "kosár", instead of "bevásárlókocsi". If I am wrong, ignore this.